src/Workspaces/Core/Portable/Workspace/Solution/Checksum_Factory.cs

```diff
@@ -14,41 +15,44 @@ namespace Microsoft.CodeAnalysis
     // all these are just helper methods
     internal partial class Checksum
     {
+        private static readonly ObjectPool<IncrementalHash> s_incrementalHashPool =
+            new ObjectPool<IncrementalHash>(() => IncrementalHash.CreateHash(HashAlgorithmName.SHA1), size: 20);
+
         public static Checksum Create(Stream stream)
         {
-            using (var hash = IncrementalHash.CreateHash(HashAlgorithmName.SHA1))
+            using (var pooledHash = s_incrementalHashPool.GetPooledObject())
+            using (var pooledBuffer = SharedPools.ByteArray.GetPooledObject())
             {
-                using (var pooledBuffer = SharedPools.ByteArray.GetPooledObject())
-                {
-                    stream.Seek(0, SeekOrigin.Begin);
+                stream.Seek(0, SeekOrigin.Begin);
+
+                var hash = pooledHash.Object;
```

Comment:
does anything need to reset this, either before/after using it?

Reply:
hash got reset when it get into the pool